### ISSUE TITLE
[Bug]:  Error: couldn't find key name in Secret catalog/catalog-db

### ISSUE THREAD

**Issue by smaruy30** [aws-samples/eks-workshop-v2#314]
### What happened?

In the [StatefulSet with EBS Volume](https://www.eksworkshop.com/docs/fundamentals/storage/ebs/statefulset-with-ebs) module, 
the catalog-mysql-ebs pod is unable to find the secret value then reports the error `Error: couldn't find key name in Secret catalog/catalog-db`.

### What did you expect to happen?

Pod recreates without any errors.

### How can we reproduce it?

```
$ pwd
/home/ec2-user/eks-workshop-v2/environment/workspace/modules/fundamentals/storage/ebs

$ kubectl apply -k .
namespace/catalog unchanged
serviceaccount/catalog unchanged
configmap/catalog unchanged
secret/catalog-db unchanged
service/catalog unchanged
service/catalog-mysql unchanged
service/catalog-mysql-ebs unchanged
deployment.apps/catalog unchanged
statefulset.apps/catalog-mysql unchanged
statefulset.apps/catalog-mysql-ebs configured

$ kubectl get pod catalog-mysql-ebs-0 -n catalog
NAME                  READY   STATUS                       RESTARTS   AGE
catalog-mysql-ebs-0   0/1     CreateContainerConfigError   0          2m20s

$ kubectl describe pod catalog-mysql-ebs-0 -n catalog | tail -n2
  Warning  Failed                  24s (x12 over 2m34s)  kubelet                  Error: couldn't find key name in Secret catalog/catalog-db
  Normal   Pulled                  24s (x11 over 2m34s)  kubelet                  Container image "public.ecr.aws/docker/library/mysql:5.7" already present on machine
```

### EKSversion

v1.23.14-eks-ffeb93d

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.9", GitCommit:"c1de2d70269039fe55efb98e737d9a29f9155246", GitTreeState:"clean", BuildDate:"2022-07-13T14:26:51Z", GoVersion:"go1.17.11", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23+", GitVersion:"v1.23.14-eks-ffeb93d", GitCommit:"96e7d52c98a32f2b296ca7f19dc9346cf79915ba", GitTreeState:"clean", BuildDate:"2022-11-29T18:43:31Z", GoVersion:"go1.17.13", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by niallthomson** [comment]
Hi @smaruy30 thanks for reporting this. We have a fix in the works already.

**Comment by smaruy30** [comment]
Hi, @niallthomson 
Thanks for the info.

> We have a fix in the works already.

In which branch of this repo, can we find the fix?
I just pulled the latest commit from the main branch shown as below,

```
$ git remote -v
origin  https://github.com/smaruy30/eks-workshop-v2.git (fetch)
origin  https://github.com/smaruy30/eks-workshop-v2.git (push)

$ pwd
/home/ec2-user/eks-workshop-v2/environment/workspace/modules/fundamentals/storage/ebs

$ git log -n1
commit c22c92fa4e8e7ef3bda2c8e7e3374f819a700192 (HEAD -> main, origin/main, origin/HEAD)
Author: Niall Thomson <thomson.niall@gmail.com>
Date:   Thu Feb 16 21:17:30 2023 -0700

    Instruct user to checkout latest tag when using own AWS account

$ git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
```

then run the same command but still see the same error, unfortunately...

```
$ reset-environment 
Resetting the environment, please wait
Waiting for application to become ready...
Environment is reset

$ pwd
/home/ec2-user/eks-workshop-v2/environment/workspace/modules/fundamentals/storage/ebs

$ kubectl apply -k .
namespace/catalog unchanged
serviceaccount/catalog unchanged
configmap/catalog unchanged
secret/catalog-db unchanged
service/catalog unchanged
service/catalog-mysql unchanged
service/catalog-mysql-ebs created
deployment.apps/catalog configured
statefulset.apps/catalog-mysql unchanged
statefulset.apps/catalog-mysql-ebs created

$ kubectl get pod catalog-mysql-ebs-0 -n catalog
NAME                  READY   STATUS                       RESTARTS   AGE
catalog-mysql-ebs-0   0/1     CreateContainerConfigError   0          29s

$ kubectl describe pod catalog-mysql-ebs-0 -n catalog | tail -n2
  Normal   Pulled                  1s (x6 over 42s)  kubelet                  Container image "public.ecr.aws/docker/library/mysql:5.7" already present on machine
  Warning  Failed                  1s (x6 over 42s)  kubelet                  Error: couldn't find key name in Secret catalog/catalog-db
```

**Comment by jonoden** [comment]
For the time being you can replace this on line 45 in `/workspace/modules/fundamentals/storage/ebs/statefulset-mysql.yaml`
```
            - name: MYSQL_DATABASE
              valueFrom:
                secretKeyRef:
                  name: catalog-db
                  key: name
```

with 
```
            - name: MYSQL_DATABASE
              value: catalog
```

**Comment by smaruy30** [comment]
@jonoden 
Thanks for the quick fix. It worked.

**Comment by niallthomson** [comment]
Fixed by #331
